gadgets/trace_capabilities/test/trace_capabilities_test.go

```diff
@@ -0,0 +1,154 @@
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-trace-capabilities"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{
+		containers.WithContainerImage(containerImage),
+		containers.WithStartAndStop(),
+	}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-trace-capabilities")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(
+		containerName,
+		"while true; do nice -n -20 echo; sleep 0.1; done",
+		containerOpts...,
+	)
+
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime)))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns)))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntries := []*traceCapabilitiesEvent{
+				{
+					CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+					Cap:        "CAP_SYS_NICE",
+					Syscall:    "SYS_SETPRIORITY",
+					Audit:      1,
+					Capable:    false,
+					Task:       "nice",
+
+					// Check the existence of the following fields
+					MountNsID:     utils.NormalizedInt,
+					Timestamp:     utils.NormalizedStr,
+					Pid:           utils.NormalizedInt,
+					Uid:           utils.NormalizedInt,
+					Gid:           utils.NormalizedInt,
+					Tgid:          utils.NormalizedInt,
+					Kstack:        utils.NormalizedStr,
+					Insetid:       utils.NormalizedInt,
+					CapEffective:  utils.NormalizedStr,
+					CurrentUserNs: utils.NormalizedInt,
+					TargetUserNs:  utils.NormalizedInt,
+				},
+			}
+
+			normalize := func(e *traceCapabilitiesEvent) {
+				utils.NormalizeCommonData(&e.CommonData)
+				utils.NormalizeInt(&e.MountNsID)
+				utils.NormalizeString(&e.Timestamp)
+				utils.NormalizeInt(&e.Pid)
+				utils.NormalizeString(&e.Kstack)
+				utils.NormalizeString(&e.CapEffective)
+
+				// Manually normalize fields that might contain 0
+				e.Uid = utils.NormalizedInt
+				e.Gid = utils.NormalizedInt
+				e.Tgid = utils.NormalizedInt
+				e.CurrentUserNs = utils.NormalizedInt
+				e.TargetUserNs = utils.NormalizedInt
+				e.Insetid = utils.NormalizedInt
```

Comment:
I don't think it makes sense to set these fields to `NormalizedInt`. The idea of this is to be able to check that a given field was present on the json, by assigning it here, that check is not possible. i.e. if the field wasn't present on the json output, the test will pass the same.

I see the following options:
1. Set all these fields to 0 and remove them from the expected entry
2. Remove them from `traceCapabilitiesEvent` so they aren't considered at all.

I think the first approach is better, it'll be able to catch if the fields exist on the json but are the wrong type.